kong/pdk/request.lua

```diff
@@ -568,10 +569,7 @@ local function new(self)
       error("header name must be a string", 2)
     end
 
-    local header_value = _REQUEST.get_headers()[name]
-    if type(header_value) == "table" then
-      return header_value[1]
-    end
+    local header_value = var["http_" .. lower(name:gsub("-", "_"))]
```

Comment:
Should we use `ngx.re.gsub` instead of `string.gsub`?

Reply:
You're replying at the speed of light!
Personally I think this can be done by simpler logic(rather than regex). Are there other alternatives to do that?

Reply:
I think it is OK, maybe I am too prefer regex.